topi/python/topi/cuda/nms.py

```diff
@@ -93,44 +67,41 @@ def get_valid_counts_ir(data, valid_count, out, out_indices,
     valid_count = ib.buffer_ptr(valid_count)
     out = ib.buffer_ptr(out)
     out_indices = ib.buffer_ptr(out_indices)
-    atomic_add_return = ib.allocate(
-        valid_count.dtype, (1,), name='atomic_add_return', scope='local')
     one_count = tvm.tir.const(1, dtype=valid_count.dtype)
     one = tvm.tir.const(1, dtype=out.dtype)
     score_threshold = tvm.ir.make_node(
         "FloatImm", dtype="float32", value=score_threshold)
     id_index = tvm.ir.make_node("IntImm", dtype="int32", value=id_index)
     score_index = tvm.ir.make_node("IntImm", dtype="int32", value=score_index)
 
-    max_threads = int(tvm.target.Target.current(
-        allow_none=False).max_num_threads)
-    nthread_tx = max_threads
-    nthread_bx = batch_size * num_anchors // max_threads + 1
```

Comment:
Would like to know more about the reason behind the change, perhaps share some benchmark numbers?  In some scenarios, like TF MaskRCNN,  a large number of boxes (`num_anchors` > 20000) are in one batch, multiple threads here might provide performance improvement.

Reply:
It seems that the gpu `get_valid_counts` does not need to moves valid boxes to the top of input data because of the `argsort` in the `nms`, so using the original `get_valid_counts_ir` ought to be better. And do you think `rearrange_indices_out_ir` is the right way to do it?

Reply:
Only using one thread will regress the performance a lot. More benchmark should be shown according to previous PRs' workloads.